src/lager_logger_api.erl

```diff
@@ -0,0 +1,215 @@
+new(RootPath, Level, _Loggers) ->
+    application:set_env(lager, log_root, RootPath),
+    application:set_env(lager, crash_log, "crash.log"),
+    application:set_env(lager, error_logger_hwm, 500),
+    error_logger:info_msg("Setup Lager Logger API~n", []),
+
+    application:set_env(lager, handlers,
+                        [{lager_file_backend, [{file, ?LOG_FILE_NAME_INFO}, {level, none},
+                                               {size, 10485760}, {date, "$H0"},
+                                               {formatter, lager_leofs_formatter},
+                                               {formatter_config, ["[", sev, "]\t", atom_to_list(node()), "\t", leodate, "\t", leotime, "\t", {module, "null"}, ":", {function, "null"}, "\t", {line, "0"}, "\t", message, "\n"]},
+                                               {rotator, leo_logger_rotator}
+                                              ]},
+                         {lager_file_backend, [{file, ?LOG_FILE_NAME_ERROR}, {level, none},
+                                               {size, 10485760}, {date, "$H0"},
+                                               {formatter, lager_leofs_formatter},
+                                               {formatter_config, ["[", sev, "]\t", atom_to_list(node()), "\t", leodate, "\t", leotime, "\t", {module, "null"}, ":", {function, "null"}, "\t", {line, "0"}, "\t", message, "\n"]},
+                                               {rotator, leo_logger_rotator}
+                                              ]}
+                        ]),
+
+    lager:start(),
+
+    {ok, Handlers} = ?log_handlers(Level),
+    lists:foreach(fun({File, FLevel}) ->
+                          lager:set_loglevel(lager_file_backend, File, FLevel)
+                  end, Handlers),
+
+    ets:new(?LOG_ID_TO_SINK_ETS,
+            [named_table, bag, public, {read_concurrency, true}]),
+    ok.
+
+-spec(new(LogGroup, LogId, RootPath, LogFileName) ->
+             ok when LogGroup::atom(),
+                     LogId::atom(),
+                     RootPath::string(),
+                     LogFileName::string()).
+new(LogGroup, LogId, _RootPath, LogFileName) ->
+    lager_app:configure_sink(LogGroup,
+                             [{handlers, [{lager_file_backend, [{file, LogFileName}, {level, info},
+                                                                {size, 10485760}, {date, "$D0"},
+                                                                {formatter, lager_default_formatter},
+                                                                {formatter_config, [message, "\n"]},
+                                                                {rotator, leo_logger_rotator}
+                                                               ]}]},
+                              {async_threshold, 500},
+                              {async_threshold_window, 50}]),
+    ets:insert(?LOG_ID_TO_SINK_ETS, {LogId, {LogGroup, LogFileName}}),
```

Comment:
those parameters also should be chosen based on the result of benchmark I suggested on the above comment. also let me confirm async_threshold(_window) are absent on the setting of the leveled error logger above. is it intentional?